### PR TITLE
Segment Tree RMQ implemented in Dart

### LINKED ISSUE #2164: Segment Tree Range Minimum Query
- [x] C - @MdAkdas
- [x] C++
- [ ] C#
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go
- [x] Java
- [ ] JS - @JanaSabuj
- [ ] Kotlin
- [ ] Php
- [x] Python
- [x] Readme - @chandrikadeb7
- [ ] Ruby
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2164 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Segment Tree is implemented in Dart for Range Minimum Queries

#### Languages Used:
- Dart

#### Files Added:
- Segment_Tree_RMQ.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Rukmini-Meda** [comment]
I have implemented Segment Trees for range minimum queries in Dart. Please review and merge. Thank you.

File: Segment_Tree_RMQ/Segment_Tree_RMQ.dart [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    
+    int q = int.parse(input);
+
+    List queries = List<int>(2);
+
+
+    // Taking input of queries.
+    print("Enter queries: ");
+
+    for(int i = 0 ; i < q ; i ++){
```
Comment (MastersAbh) [inline Segment_Tree_RMQ/Segment_Tree_RMQ.dart:126]: space after for

Comment (Rukmini-Meda) [reply Segment_Tree_RMQ/Segment_Tree_RMQ.dart:126]: Do I have to give space after if as well?

**Comment by Rukmini-Meda** [comment]
@MastersAbh I have implemented the update node function for segment tree. Please take a look and review. Thank you.

**Comment by MastersAbh** [comment]
Please squash your commits @Rukmini-Meda

**Comment by Rukmini-Meda** [comment]
I have squashed the commits. Please merge.

**Pull request merged**